### PR TITLE
Hotfix/464 broken anchor links

### PR DESCRIPTION
**Pull request by pete-hotchkiss**
has fixes for broken anchor references when annotation link to items in other @group's

### CODE REVIEW

**Comment by hyperbrave** [comment]
What needs to happen to merge this pull?

**Comment by HugoGiraudel** [comment]
A review from maintainers. :D

Ping @SassDoc/owners.

**Comment by pascalduez** [comment]
I tried to wrap my head around this some time ago, and was not convinced this is the right way, there's a missing brick on why it is this way.

**Comment by pete-hotchkiss** [comment]
been a few months since I wrote this code, but I recall being aware my solution was somewhat 'baked' - i.e effectivly baking a new property into the annotation object is far from elegant. Would be more preferable to have a reference to the parent scope so we could access the grouping property directly rather than adding/removing as we go.

I was also aware I didn't want to completely break existing theme templates...

My stuff is ugly, but it works - and it got my documentation working with a locally patched version of the library.

**Comment by HugoGiraudel** [comment]
Hey I’m kind of busy right now, but this is on my radar. I’ll try to check this next week.

File: src/annotation/annotations/alias.js [hunk 1]
```diff
@@ -27,7 +28,14 @@ export default function alias(env) {
           aliasedItem.aliased = [];
         }
 
+        if (!Array.isArray(aliasedItem.aliasedgroup)) {
```
Comment (HugoGiraudel) [inline src/annotation/annotations/alias.js]: We should probably use `aliasedGroup` to respect camelCase conventions?

**Comment by HugoGiraudel** [comment]
@SassDoc/owners The code looks simply enough to me. Can someone review it as well? I might lack a bit of context on this part of the code base.

File: src/annotation/annotations/alias.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ export default function alias(env) {
 
         let alias = item.alias;
         let name = item.context.name;
+        var aliasGroup = item.group;
```
Comment (valeriangalliat) [inline src/annotation/annotations/alias.js]: `let`

**Comment by pete-hotchkiss** [comment]
I'm out and about today, but I'll make the changes to the camelCasing this
evening and update the PR

**Comment by LeaveAirykson** [comment]
Is there any progress on this request? the problem still exists..

**Comment by pascalduez** [comment]
There's a few changes requested though.

**Comment by LeaveAirykson** [comment]
I understand that there are many requests and i didn't mean to be impatient. It's just that this pull request is open for almost 2 years now and no further progress was reported. just wanted to know if there are any news because we handle large projects with sassdoc and grouping is necessary to keep our documentary organized and this issue makes the linking kind of useless.

**Comment by d13** [comment]
> I understand that there are many requests and i didn't mean to be impatient. It's just that this pull request is open for almost 2 years now and no further progress was reported. just wanted to know if there are any news because we handle large projects with sassdoc and grouping is necessary to keep our documentary organized and this issue makes the linking kind of useless.

Same for me, I heavily rely on groups in a very large codebase.

What can I do to help get this over the finish line? Is the hold-up just camel-casing and changing a `var` to `let`?

**Comment by KittyGiraudel** [comment]
Hello there. 👋 

I am no longer maintaining SassDoc, nor do I have time to invest in that project anymore. Unfortunately, I don’t think we have dedicated maintainers to take over, so the project is kind of frozen in time and place. I‘m sorry about this.

**Comment by valeriangalliat** [comment]
Hey! On top of the PR comments, a quick rebasing / conflicts fixing needed to be done. I've pushed an up-to-date version of this PR in https://github.com/SassDoc/sassdoc/pull/567.

@d13 any chance you can checkout the code from #567 and run it against your codebase to confirm this fixes the issue for you as well? Something like `cd node_modules; rm -rf sassdoc; git clone --branch hotfix/464-broken-anchor-links https://github.com/SassDoc/sassdoc; cd sassdoc; yarn; make` should do.

Since this was opened as a `hotfix` and I approved it back then without any concern about breaking changes, I believe it would make sense to release this with a patch version. Do that sound reasonable @KittyGiraudel?

Cheers!

**Comment by KittyGiraudel** [comment]
Definitely. Happy to release a hotfix once it‘s merged.

**Comment by valeriangalliat** [comment]
Released with sassdoc@2.7.4 and sassdoc-theme-default@2.8.5!

**Pull request closed without merging**